### ISSUE TITLE
DPI setting not working correctly on Logitech G300s

### ISSUE THREAD

**Issue by KleberPF** [libratbag/libratbag#1216]
**Information**

- `ratbagd` version (`ratbagd --version`): 0.16
- Distribution: Pop!_OS 21.04
- Kernel version (ex. `uname -srmo`): `Linux 5.13.0-7614-generic x86_64 GNU/Linux`

Device info (if applicable):
```
$ ratbagctl Logi info
warbling-mara - Logitech G300s Optical Gaming Mouse
             Model: usb:046d:c246:0
 Number of Buttons: 9
    Number of Leds: 1
Number of Profiles: 3
Profile 0: (active)
  Name: n/a
  Report Rate: 500Hz
  Resolutions:
    0: 50dpi
    1: 400dpi (active) (default)
    2: 450dpi
    3: 700dpi
  Button: 0 is mapped to 'button 1'
  Button: 1 is mapped to 'button 2'
  Button: 2 is mapped to 'button 3'
  Button: 3 is mapped to macro '↕SCROLLLOCK'
  Button: 4 is mapped to 'button 5'
  Button: 5 is mapped to macro '↓LEFTCTRL ↕RESERVED ↑LEFTCTRL'
  Button: 6 is mapped to macro '↓LEFTALT ↕RESERVED ↑LEFTALT'
  Button: 7 is mapped to 'profile-cycle-up'
  Button: 8 is mapped to 'resolution-cycle-up'
  LED: 0, depth: rgb111, mode: on, color: 00ffff
Profile 1:
  Name: n/a
  Report Rate: 1000Hz
  Resolutions:
    0: 100dpi
    1: 200dpi (active) (default)
    2: 300dpi
    3: 500dpi
  Button: 0 is mapped to 'button 1'
  Button: 1 is mapped to 'button 2'
  Button: 2 is mapped to 'button 3'
  Button: 3 is mapped to 'button 4'
  Button: 4 is mapped to 'button 5'
  Button: 5 is mapped to 'resolution-down'
  Button: 6 is mapped to 'resolution-up'
  Button: 7 is mapped to 'profile-cycle-up'
  Button: 8 is mapped to 'resolution-alternate'
  LED: 0, depth: rgb111, mode: on, color: ffffff
Profile 2:
  Name: n/a
  Report Rate: 500Hz
  Resolutions:
    0: 100dpi
    1: 400dpi (active) (default)
    2: 300dpi
    3: 800dpi
  Button: 0 is mapped to 'button 1'
  Button: 1 is mapped to 'button 2'
  Button: 2 is mapped to 'button 3'
  Button: 3 is mapped to macro '↕SCROLLLOCK'
  Button: 4 is mapped to 'button 5'
  Button: 5 is mapped to macro '↓LEFTCTRL ↕C ↑LEFTCTRL'
  Button: 6 is mapped to macro '↓LEFTCTRL ↕V ↑LEFTCTRL'
  Button: 7 is mapped to 'profile-cycle-up'
  Button: 8 is mapped to 'resolution-cycle-up'
  LED: 0, depth: rgb111, mode: on, color: 00ffff
```

**Describe the bug**
I'm not able to set the DPI correctly. Some values are bugged (for example, 850 is very slow, but 900 isn't) and I'm not sure if any of them is correct. I set it to 400 DPI and booted up Minecraft with the same sensitivity I use on Windows (with raw input on) and it is WAY faster.

**To Reproduce**
1. Run `ratbagctl Logi dpi set 850`
2. Move the cursor around and notice that it's very slow
3. Run `ratbagctl Logi dpi set 900`
4. Move the cursor around and notice that it's faster than with 900 DPI set

**Logs**
Please start the daemon with verbosity and reproduce the issue.

First make sure it isn't running already.
```
$ ratbagd --verbose=raw
ratbag debug: New device: SIGMACHIP USB Keyboard
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: No data file found for 1c4f:0002
ratbag debug: New device: SIGMACHIP USB Keyboard
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: No data file found for 1c4f:0002
ratbag debug: New device: Logitech G300s Optical Gaming Mouse
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: device assigned driver logitech_g300
ratbag debug: hidraw info: bus 0x03 vendor 0x46d product 0xffffc246
ratbag debug: Logitech G300s Optical Gaming Mouse is device '/dev/hidraw0'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
ratbag debug: hidraw info: bus 0x03 vendor 0x46d product 0xffffc246
ratbag debug: Logitech G300s Optical Gaming Mouse is device '/dev/hidraw1'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
ratbag debug: - HID report ID 01
ratbag debug: - HID report ID 80
ratbag debug: - HID report ID f0
ratbag debug: - HID report ID f1
ratbag debug: - HID report ID f2
ratbag debug: - HID report ID f3
ratbag debug: - HID report ID f4
ratbag debug: - HID report ID f5
ratbag raw: feature get:   f3 06 03 01 88 09 0e 40 01 00 00 02 00 00 03 00 00 00 00 47 05 00 00 00 01 00 00 04 00 0d 00 00 0c 00 00
ratbag raw: feature get:   f4 07 00 02 84 06 0a 02 01 00 00 02 00 00 03 00 00 04 00 00 05 00 00 0b 00 00 0a 00 00 0d 00 00 0e 00 00
ratbag raw: feature get:   f5 06 03 02 88 06 10 44 01 00 00 02 00 00 03 00 00 00 00 47 05 00 00 00 01 06 00 01 19 0d 00 00 0c 00 00
ratbag raw: feature get:   f0 02 00 00
ratbag raw: 'Logitech G300s Optical Gaming Mouse' is in profile 0
ratbag debug: driver match found: Logitech G300
hidraw0: "Logitech G300s Optical Gaming Mouse", 3 profiles
ratbag debug: New device: Logitech G300s Optical Gaming Mouse
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: device assigned driver logitech_g300
ratbag debug: hidraw info: bus 0x03 vendor 0x46d product 0xffffc246
ratbag debug: Logitech G300s Optical Gaming Mouse is device '/dev/hidraw0'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
DBus server ready
ratbag debug: Profile 0 changed, rewriting
ratbag raw: feature set:   f0 42 00 00
ratbag raw: feature set:   f3 06 03 01 92 09 0e 40 01 00 00 02 00 00 03 00 00 00 00 47 05 00 00 00 01 00 00 04 00 0d 00 00 0c 00 00
ratbag debug: Profile 1 changed, rewriting
ratbag raw: feature set:   f4 07 00 02 84 06 0a 02 01 00 00 02 00 00 03 00 00 04 00 00 05 00 00 0b 00 00 0a 00 00 0d 00 00 0e 00 00
ratbag debug: Profile 2 changed, rewriting
ratbag raw: feature set:   f5 06 03 02 88 06 10 44 01 00 00 02 00 00 03 00 00 00 00 47 05 00 00 00 01 06 00 01 19 0d 00 00 0c 00 00
ratbag debug: Profile 0 changed, rewriting
ratbag raw: feature set:   f0 42 00 00
ratbag raw: feature set:   f3 06 03 01 91 09 0e 40 01 00 00 02 00 00 03 00 00 00 00 47 05 00 00 00 01 00 00 04 00 0d 00 00 0c 00 00
ratbag debug: Profile 0 changed, rewriting
ratbag raw: feature set:   f0 42 00 00
ratbag raw: feature set:   f3 06 03 01 88 09 0e 40 01 00 00 02 00 00 03 00 00 00 00 47 05 00 00 00 01 00 00 04 00 0d 00 00 0c 00 00
```

**Additional context**
The debug output is after I ran the following commands:
```
$ ratbagctl Logi dpi set 900
$ ratbagctl Logi dpi get
900dpi
$ ratbagctl Logi dpi set 850
$ ratbagctl Logi dpi get
850dpi
$ ratbagctl Logi dpi set 400
```

**Comment by staticssleever668** [comment]
It looks like related piece of code was changed in #1035. Would it be possible for one of you to revert the change and check if it fixes the problem?
`git revert --no-commit 20b573668e664bb251db3bb1d078255085cebf98`
ratslap - utility for G300 mentioned in #1458 - seems to set DPI the way before #1035 was merged.

**Comment by thalting** [comment]
> It looks like related piece of code was changed in #1035. Would it be possible for one of you to revert the change and check if it fixes the problem? `git revert --no-commit 20b573668e664bb251db3bb1d078255085cebf98` ratslap - utility for G300 mentioned in #1458 - seems to set DPI the way before #1035 was merged.

thanks!
works fine now.
